web3/_utils/events.py

```diff
@@ -102,6 +99,13 @@
     )
 
 
+@curry
```

Comment:
nit: There's probably no need to curry this, right? Doesn't seem like this method is being used in that context.